lib/rules/sort-imports.js

```diff
@@ -83,62 +154,139 @@ module.exports = {
+                }
+            } else if (reactRegExp.test(nodeASource)) {
+                return -1;
+            } else if (reactRegExp.test(nodeBSource)) {
+                return 1;
             }
+
+            // usedMemberSyntax is different
+            const nodeAGroupOrder = getMemberParameterGroupIndex(nodeA);
+            const nodeBGroupOrder = getMemberParameterGroupIndex(nodeB);
+
+            if (nodeAGroupOrder < nodeBGroupOrder) {
+                return -1;
+            }
+            if (nodeAGroupOrder > nodeBGroupOrder) {
+                return 1;
+            }
+
+            // usedMemberSyntax is the same
+            let sourceCodeA = sourceCode.getText(nodeA);
+            let sourceCodeB = sourceCode.getText(nodeB);
+
+            if (ignoreCase) {
+                sourceCodeA = sourceCodeA.toLowerCase();
+                sourceCodeB = sourceCodeB.toLowerCase();
+            }
+
+            if (sourceCodeA < sourceCodeB) {
+                return -1;
+            }
+            if (sourceCodeA > sourceCodeB) {
+                return 1;
+            }
+            return 0;
+        }
+
+        /**
+         * Fixes declaration order
+         *
+         * @param {fixer} fixer - fixer the RuleFixer
+         * @returns {*} - fixer result
+         */
+        function declarationsFixerFunction(fixer) {
+            const replaceOutput = [];
+
+            for (let i = 0; i < declarations.length; i++) {
+                replaceOutput.push(sourceCode.getText(declarations[i]));
+            }
+
+            return fixer.replaceTextRange(
+                [firstDeclaration.range[0], lastDeclaration.range[1]],
+                replaceOutput.join("\n")
+            );
         }
 
         return {
             ImportDeclaration: function(node) {
-                if (previousDeclaration) {
-                    const currentMemberSyntaxGroupIndex = getMemberParameterGroupIndex(node),
-                        previousMemberSyntaxGroupIndex = getMemberParameterGroupIndex(previousDeclaration);
-                    let currentLocalMemberName = getFirstLocalMemberName(node),
-                        previousLocalMemberName = getFirstLocalMemberName(previousDeclaration);
+                firstDeclaration = firstDeclaration || node;
+                lastDeclaration = node;
 
-                    if (ignoreCase) {
-                        previousLocalMemberName = previousLocalMemberName && previousLocalMemberName.toLowerCase();
-                        currentLocalMemberName = currentLocalMemberName && currentLocalMemberName.toLowerCase();
+                declarations.push(node);
+                const sortedDeclarations = declarations.slice().sort(sortingFunction);
+
+                if (declarations.length > 1) {
+                    previousDeclaration = declarations[declarations.length - 2];
+                }
+
+                let sorted = true;
+
+                declarations.forEach((importNode, ind) => {
```

Comment:
Consider using `some()` or `every()` (the `return false;` in your callback does not terminate the `forEach()` execution).